src/main/java/org/terasology/core/world/generator/facets/SurfaceRoughnessFacet.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.core.world.generator.facets;
+
+import org.terasology.biomesAPI.Biome;
+import org.terasology.math.Region3i;
+import org.terasology.world.generation.Border3D;
+import org.terasology.world.generation.facets.base.BaseFieldFacet2D;
+import org.terasology.world.generation.facets.base.BaseObjectFacet2D;
+
+/**
+ * Modifies the amount of noise to be added to the surface shape, to determine the amount of lumps and cliffs and things.
+ */
+public class SurfaceRoughnessFacet extends BaseFieldFacet2D {
```

Comment:
As this is a 2d facet, how does this play along with having multiple surfaces at the same coordinate? Is this rather a `ElevationRoughnessFacet`?

Reply:
In the cases where there are multiple surfaces at the same 2D coordinates, they are (at least with the default world generator) not too far apart, and the SurfaceRoughnessFacet varies rather slowly, so the fact that it's 2D isn't a problem.

Reply:
What does "not too far apart" mean? And aren't surfaces supposed to also be used for large cavernous areas where some world generator could spawn underground vegetation and structures? Or floating islands being generated way up in the sky?

Reply:
If a world generator had multiple surfaces on a large scale, to the extent that this became important, that would already be breaking a lot of other assumptions made by this set of facet providers (e.g. the temperature, humidity and biome facets would also have to be 3D in such a world generator, and the ElevationFacet itself wouldn't really be applicable). I don't think that's really worth preparing for unless it's actually likely to happen. Making this a 3D facet would make it substantially more computationally expensive. In the case of smaller features like those provided by the Caves and FlyingIslands modules, they would be generated after the DensityNoiseProvider anyway, so the SurfaceRoughnessFacet wouldn't actually affect them.

To be more precise, the DensityNoiseProvider won't displace the surface by more than about 40 blocks vertically, except very rarely. The gradient of the SurfaceRoughnessFacet generally wouldn't exceed 1/300 vertically (and its values are roughly in the range -1 to 2). If I just changed the SimplexRoughnessProvider so that it varied vertically too, it would take a rare coincidence for the SurfaceRoughnessFacet's values to differ by as much as 1/4 at different surfaces within a vertical column. The difference would be noticable if you compare the same world side-by-side with the different algorithms, but it wouldn't lead to much overall change in the sort of features present.